### PR TITLE
Adjust the DPAS instruction order in TritonIntelGPUToLLVM

### PR DESCRIPTION
**Pull request by chengjunlu**
Adjust the order of the generated DPAS instruction to get better DPAS operands locality in the generated LLVM IR.

The new logic generates a large set of the DPAS instructions which can reuse the A or B operands across multiple DPAS instructions.

Add a new env-var  `TRITON_INTEL_AGGRESSIVE_DPAS_REUSE` to generate most aggressive DPAS instruction order for experimental. Will make the aggressive order as default when the IGC scalar backend could perfectly generate the best performance kernel in instruction scheduling.

### CODE REVIEW

**Review by victor-eds — APPROVED** [review]

**Comment by etiotto** [comment]
@chengjunlu I have added lambda functions to refactor the loop. The codegen is the same.

**Review by etiotto — APPROVED** [review]

File: include/triton/Tools/Sys/GetEnv.hpp [hunk 1]
```diff
@@ -37,7 +37,8 @@ inline const std::set<std::string> CACHE_INVALIDATING_ENV_VARS = {
     "TRITON_INTEL_ENABLE_FIRST_LOAD_TO_SLM",
     "TRITON_INTEL_ENABLE_INSTR_SCHED",
     "TRITON_INTEL_ENABLE_POST_PROCESS_LLIR",
-    "TRITON_INTEL_REDUCE_TRANSPOSE"
+    "TRITON_INTEL_REDUCE_TRANSPOSE",
+    "TRITON_INTEL_AGGRESSIVE_DPAS_REUSE"
```
Comment (whitneywhtsang) [inline include/triton/Tools/Sys/GetEnv.hpp:41]: alphabetical order?

**Review by whitneywhtsang — APPROVED** [review]

**Comment by etiotto** [comment]
Performance run on microbenchmarks: http://benchmarks.glados.intel.com/d/1pXX4hUSz/microbenchmarks?orgId=1&var-tag=ci%7Creorder_dpas_gen&var-bench=All&var-device=Intel%28R%29%20Data%20Center%20GPU%20Max%201550&var-compiler=triton&var-backend=All&var-baseline_backend=triton-ci-XPU%201550&var-target_backend=triton-ci-XPU%201550

Results are essentially on par with CI.

File: include/triton/Tools/Sys/GetEnv.hpp (hunk 1 above)
Comment (etiotto) [reply include/triton/Tools/Sys/GetEnv.hpp:41]: done

**Pull request merged**